auto_nag/scripts/assignee_no_login.py

```diff
@@ -53,68 +50,53 @@ def columns(self):
         if self.people.is_mozilla(assignee):
             return None
 
-        bugid = str(bug["id"])
-
-        # Avoid to ni everyday...
-        if self.has_bot_set_ni(bug):
-            do_needinfo = False
+        prod = bug["product"]
+        comp = bug["component"]
+        default_assignee = self.default_assignees[prod][comp]
+        autofix = {"assigned_to": default_assignee}
 
         # Avoid to ni if the bug has low priority and low severity.
         # It's not paramount for triage owners to make an explicit decision here, it's enough for them
         # to receive the notification about the unassignment from Bugzilla via email.
-        elif (
+        if (
             bug["priority"] not in HIGH_PRIORITY
             and bug["severity"] not in HIGH_SEVERITY
         ):
-            do_needinfo = False
-
-        else:
-            do_needinfo = True
-
-        if do_needinfo:
-            if not self.add_auto_ni(
-                bugid,
-                {
-                    "mail": bug["triage_owner"],
-                    "nickname": bug["triage_owner_detail"]["nick"],
-                },
-            ):
-                # Don't unassign if we can't needinfo.
-                return None
+            needinfo = None
+            autofix["comment"] = {
+                "body": f"The bug assignee didn't login in Bugzilla in the last { self.nmonths } months, so the assignee is being reset."
+            }
         else:
-            if (
-                self.unassign_count[bug["triage_owner"]]
-                >= self.max_unassign_per_triage_owner
-            ):
-                return None
-            self.unassign_count[bug["triage_owner"]] += 1
-
-        data[bugid] = {"triage_owner": bug["triage_owner_detail"]["real_name"]}
-        prod = bug["product"]
-        comp = bug["component"]
-        default_assignee = self.default_assignees[prod][comp]
-        self.autofix_assignee[bugid] = {"assigned_to": default_assignee}
-        if do_needinfo:
             reason = []
             if bug["priority"] in HIGH_PRIORITY:
                 reason.append("priority '{}'".format(bug["priority"]))
             if bug["severity"] in HIGH_SEVERITY:
                 reason.append("severity '{}'".format(bug["severity"]))
-            self.extra_ni[bugid] = {
-                "reason": "/".join(reason),
-            }
-        else:
-            self.autofix_assignee[bugid]["comment"] = {
-                "body": f"The bug assignee didn't login in Bugzilla in the last { self.nmonths } months, so the assignee is being reset."
+
+            needinfo = {
+                "mail": bug["triage_owner"],
+                "nickname": bug["triage_owner_detail"]["nick"],
+                "extra": {"reason": "/".join(reason)},
             }
 
+        self.add_prioritized_action(bug, bug["triage_owner"], needinfo, autofix)
+
+        bugid = str(bug["id"])
+        data[bugid] = {"triage_owner": bug["triage_owner_detail"]["real_name"]}
```

Comment:
If we don't do anything for a specific bug, adding the bug to `data` might cause an error (IIRC I've seen that).
Is a dry run not giving any problem?

Reply:
Since the value of `exclude_no_action_bugs()` for this tool is `True`, if we don't do anything for a specific bug, it will be removed from `data` in the next step:
https://github.com/mozilla/relman-auto-nag/blob/45080abc59808c7a502141a99fad2c4d5d69486a/auto_nag/bzcleaner.py#L309-L310

 In the dry-run it worked fine even when value of `exclude_no_action_bugs()` is `False`.

Reply:
OK, perfect. Just wanted to make sure you did dry run it.